analyzer/tools/build-logger/src/ldlogger-tool-gcc.c

```diff
@@ -406,26 +307,56 @@ int loggerGccParserCollectActions(
        * from the flag by a space character.
        * 2 == strlen("-I") && 8 == strlen("-isystem")
        */
-      if (current[1] == 'I')
+      if (strstr(current, "-I") == current)
         lastIncPos = action->arguments.size + (current[2] ? 0 : 1);
       else if (strstr(current, "-isystem") == current)
         lastSysIncPos = action->arguments.size + (current[8] ? 0 : 1);
 
       /* Determine the programming language based on -x flag.
        */
-      else if (strcmp(current, "-x") == 0)
+      else if (strstr(current, "-x") == current)
       {
         /* TODO: The language value after -x can be others too. See the man
          * page of GCC.
          * TODO: According to a GCC warning the -x flag has no effect when it
          * is placed after the last input file to be compiled.
          */
-        const char* l = argv_[i + 1];
+        const char* l = current[2] ? current[2] : argv_[i + 1];
         if (strcmp(l, "c") == 0 || strcmp(l, "c-header") == 0)
           lang = C;
         else if (strcmp(l, "c++") == 0 || strcmp(l, "c++-header") == 0)
           lang = CPP;
       }
+
+      /* Determining the output of the build command. In some cases some .o or
+       * or other files may be considered a source file when they are a
+       * parameter of a flag other than -o, such as -MT. The only usage of
+       * collecting action->output is to remove it from action->sources later.
+       */
+      else if (strstr(current, "-o") == current)
+      {
+        loggerFileInitFromPath(
+          &action->output,
+          current[2] ? current[2] : argv_[i + 1]);
+      }
+    }
+    else
```

Comment:
There are a couple of g++ options which require a file parameter:
* `-include file`
* `-imacros file`
* `-iprefix file`
* `-aux-info filename`
* `-MF file`

We will mark the parameter of these options as source files. Do not we need to exclude these files from the source file list?

Reply:
These flags are not followed by a source file (i.e. non-header C/C++ source file). So I think there is no need to handle them separately.